app/src/test/java/org/simple/clinic/registration/phone/RegistrationPhoneScreenControllerTest.kt

```diff
@@ -280,4 +289,62 @@ class RegistrationPhoneScreenControllerTest {
 
     verify(screen, times(2)).hideAnyError()
   }
+
+  @Test
+  @Parameters(method = "params for showing logged out dialog")
+  fun `when the screen is created and a local logged in user exists, show the logged out dialog if the user is unauthorized`(
+      loggedInUser: Optional<User>,
+      shouldShowLoggedOutDialog: Boolean
+  ) {
+    whenever(userSession.loggedInUser()).thenReturn(Observable.just(loggedInUser))
+
+    uiEvents.onNext(RegistrationPhoneScreenCreated())
+
+    if (shouldShowLoggedOutDialog) {
+      verify(screen).showLoggedOutOnThisDeviceDialog()
+    } else {
+      verify(screen, never()).showLoggedOutOnThisDeviceDialog()
+    }
+  }
+
+  @Suppress("Unused")
+  fun `params for showing logged out dialog`(): List<List<Any>> {
+    fun testCase(
+        loggedInUser: User?,
+        shouldShowLoggedOutDialog: Boolean
+    ): List<Any> {
+      return listOf(loggedInUser.toOptional(), shouldShowLoggedOutDialog)
+    }
+
+    return listOf(
+        testCase(
+            loggedInUser = null,
+            shouldShowLoggedOutDialog = false
+        ),
+        testCase(
+            loggedInUser = PatientMocker.loggedInUser(loggedInStatus = NOT_LOGGED_IN),
+            shouldShowLoggedOutDialog = false
+        ),
+        testCase(
+            loggedInUser = PatientMocker.loggedInUser(loggedInStatus = OTP_REQUESTED),
+            shouldShowLoggedOutDialog = false
+        ),
+        testCase(
+            loggedInUser = PatientMocker.loggedInUser(loggedInStatus = LOGGED_IN),
+            shouldShowLoggedOutDialog = false
+        ),
+        testCase(
+            loggedInUser = PatientMocker.loggedInUser(loggedInStatus = RESETTING_PIN),
+            shouldShowLoggedOutDialog = false
+        ),
+        testCase(
+            loggedInUser = PatientMocker.loggedInUser(loggedInStatus = RESET_PIN_REQUESTED),
+            shouldShowLoggedOutDialog = false
+        ),
+        testCase(
+            loggedInUser = PatientMocker.loggedInUser(loggedInStatus = UNAUTHORIZED),
```

Comment:
Another reason for moving the check for whether a user is `UNAUTHORIZED` or not.
The `RegistrationPhoneScreenControllerTest` need not be aware of various `loggedInStatus`.
This logic should not be part of the Controller itself.

Reply:
Yup. Raised a story to move this out at some point: https://www.pivotaltracker.com/story/show/166607823

Reply:
Done.